synapse/storage/databases/main/devices.py

```diff
@@ -302,11 +304,17 @@ async def get_device_updates_by_remote(
 
             last_processed_stream_id = update_stream_id
 
+        # len(query_map) + len(cross_signing_keys_by_user) <= len(updates) here,
+        # so len(query_map) + len(cross_signing_keys_by_user) <= limit.
+
         results = await self._get_device_update_edus_by_remote(
             destination, from_stream_id, query_map
         )
 
-        # add the updated cross-signing keys to the results list
+        # len(results) <= len(query_map) here,
```

Comment:
Agreed, but I had to look at your comment on _get_device_update_edus_by_remote for this

Reply:
Do you think I should add some words?

Reply:
Nah, I think it's okay!